recipes/linux-headers-generic/all/conanfile.py

```diff
@@ -0,0 +1,38 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class LinuxHeadersGenericConan(ConanFile):
+    name = "linux-headers-generic"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.kernel.org/"
+    license = "GPL-2.0-only"
+    description = "Generic Linux kernel headers"
+    topics = ("linux", "headers", "generic")
+    settings = "os", "arch"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("linux-headers-generic supports only Linux")
+        if tools.cross_building(self):
+            raise ConanInvalidConfiguration("linux-headers-generic can not be cross-compiled")
+
```

Comment:
I think you can add a `package_id` method to declare the package header only.

Reply:
doesn't header only mean "same package for any combination of settings" ? In this case, headers are architecture-dependent. Also, what do we gain by doing that?

Reply:
IMHO, I prefer to have one package per arch if headers are different than a single package with headers for different architectures.

Reply:
right now, it's same package for all architectures, right?
as I can see, it just copies everything: https://github.com/conan-io/conan-center-index/pull/6821/files#diff-009916074cbd7649ac284de4ed2298095d3f23dd1c3e7a6b03e756ddb98c35e3R38

Reply:
No. `make headers` detects the architecture and installs the corresponding headers. Also, `ARCH=something` can be passed to specify that we want the architecture-dependent headers for `something`

Reply:
But OS at least could be removed. It's only related to LInux, and Make only differs based architecture.

Reply:
Keep `os` in settings attribute, but remove from the package ID:

```
def package_id(self):
    del self.info.settings.os
```